clients/src/main/java/org/apache/kafka/clients/consumer/internals/Heartbeat.java

```diff
@@ -32,6 +32,7 @@
     private final Timer pollTimer;
 
     private volatile long lastHeartbeatSend = 0L;
+    private RequestFuture<Void> heartbeatFuture = null;
```

Comment:
Do we need a reference to the future? A boolean would probably be good enough.

Reply:
I have a follow-up PR which leverages on the future which is based on this one --- I think for this PR it does not harm to keep the future object itself although it is only used for boolean checks in unit tests, wdyt?